public/app/features/alerting/unified/components/rule-viewer/tabs/components/VersionHistoryTable.tsx

```diff
@@ -0,0 +1,185 @@
+import { useState } from 'react';
+
+import { dateTimeFormat, dateTimeFormatTimeAgo } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Badge, Button, Checkbox, Column, ConfirmModal, InteractiveTable, Stack, Text } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
+import { computeVersionDiff } from 'app/features/alerting/unified/utils/diff';
+import { DiffGroup } from 'app/features/dashboard-scene/settings/version-history/DiffGroup';
+import { Diffs, jsonDiff } from 'app/features/dashboard-scene/settings/version-history/utils';
+import { GrafanaRuleDefinition, RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { UpdatedByUser } from './UpdatedBy';
+
+const VERSIONS_PAGE_SIZE = 20;
+
+export function VersionHistoryTable({
+  onVersionsChecked,
+  ruleVersions,
+  disableSelection,
+  checkedVersions,
+}: {
+  onVersionsChecked(id: string): void;
+  ruleVersions: Array<RulerGrafanaRuleDTO<GrafanaRuleDefinition>>;
+  disableSelection: boolean;
+  checkedVersions: Map<string, boolean>;
```

Comment:
nit: I'd try to use `Set<string>` instead of map. Instead of checking the boolean flag, just the presence of a version would mean it is checked. Nothing blocking though

Reply:
done!